src/TheSpace/TheSpace.sol

```diff
@@ -112,15 +114,58 @@ contract TheSpace is HarbergerMarket {
             return new uint256[](0);
         }
         uint256 left = total - offset;
-        uint256 pageSize = left > limit ? limit : left;
+        uint256 size = left > limit ? limit : left;
 
-        uint256[] memory tokens = new uint256[](pageSize);
+        uint256[] memory tokens = new uint256[](size);
 
-        for (uint256 i = 0; i < pageSize; i++) {
+        for (uint256 i = 0; i < size; i++) {
             uint256 tokenIndex = i + offset;
             tokens[i] = tokenOfOwnerByIndex(owner, tokenIndex);
         }
 
         return tokens;
     }
+
+    /**
+     * @notice Get owned pixels for a user using pagination.
+     *
+     * @dev use getTokensByOwner internally
+     * @return total owned pixel total amount for this user
+     * @return size page size
+     * @return page page number
+     * @return pixels pixels in this page
+     */
+    function getPixelsPageByOwner(
+        address owner_,
+        uint256 size_,
+        uint256 page_
```

Comment:
I think we should make `getPixelsPageByOwner` and `getTokensByOwner` share the same pagination design, `size` + `offset`.

Reply:
For naming, maybe  the `getTokensByOwner` one can be kept .  `ERC721Enumerable` have methods like  `tokenOfOwnerByIndex`, `tokenByIndex`, which return `tokenId`.

Reply:
> For naming, maybe the `getTokensByOwner` one can be kept . `ERC721Enumerable` have methods like `tokenOfOwnerByIndex`, `tokenByIndex`, which return `tokenId`.

or just remove this method,  the element of `TheSpace` is `Pixel`, not `token`

Reply:
It's good to move to `HarbergerMarket.sol`

Reply:
So you think that `getTokensByOwner` is needed?

Reply:
Though more gas is used on deployment, I think it would be nice if we can keep it since it's much lower gas than `getPixelsByOwner`, and other developers may use it.

Reply:
> Though more gas is used on deployment, I think it would be nice if we can keep it since it's much lower gas than `getPixelsByOwner`, and other developers may use it.

Are you saying the cases that  `getTokensByOwner` is included in transactions (for example,  called by other contracts)?  In such cases,  I think developers can use the methods provided by `ERC721Enumerable` to do the job ?

Reply:
Oh, you are right, `getTokensByOwner` is convenient for client-side, not really for the contract. Please feel free to remove it, thanks!